guides/plugins/plugins/checkout/cart/tax-provider.md

```diff
@@ -0,0 +1,153 @@
+
+{% code title="<plugin root>/src/Checkout/Cart/Tax/TaxProvider.php" %}
+
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Checkout\Cart\Tax;
+
+use Shopware\Core\Checkout\Cart\Cart;
+use Shopware\Core\Checkout\Cart\LineItem\LineItem;
+use Shopware\Core\Checkout\Cart\Tax\Struct\CalculatedTax;
+use Shopware\Core\Checkout\Cart\Tax\Struct\CalculatedTaxCollection;
+use Shopware\Core\Checkout\Cart\TaxProvider\AbstractTaxProvider;
+use Shopware\Core\Checkout\Cart\TaxProvider\Struct\TaxProviderStruct;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use TaxJar\Client;
+
+class TaxProvider extends AbstractTaxProvider
+{
+    public function provideTax(Cart $cart, SalesChannelContext $context): TaxProviderStruct
+    {
+        $lineItemTaxes = [];
+    
+        foreach ($cart->getLineItems() as $lineItem) {
+            $taxRate = 50;
+            $price = $lineItem->getPrice()->getTotalPrice();
+            $tax = $price * $taxRate / 100;
+        
+            $lineItemTaxes[$lineItem->getId()] = new CalculatedTaxCollection(
+                [
+                    new CalculatedTax($tax, $taxRate, $price),
+                ]
+            );
+        }
+        
+        // you could do the same for deliveries
+        // $deliveryTaxes = [];
+        
+        // foreach ($cart->getDeliveries() as $delivery) {
+        // ...
+        // }
+        
+        // If you call a tax provider, you will probably get calculated tax sums for the whole cart
+        // Use the cartPriceTaxes to let Shopware show the correct sums in the checkout
+        // If omitted, Shopware will try to calculate the tax sums itself
+        // $cartPriceTaxes = [];
+        
+        return new TaxProviderStruct(
+            $lineItemTaxes,
+            // $deliveries,
+            // $cartPriceTaxes
+        );
+    }
+}
+
+```
+
+{% endcode %}
+
+## Registering the tax provider in the DI container
+
+If you've created your tax provider, you need to register it in the DI container. To do so, you need to create a new service in the `services.xml` file and tag the service as `shopware.tax_provider`.
+
+{% code title="<plugin root>/src/Resources/config/services.xml" %}
+
+```markup
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <service id="SwagTaxProviders\TaxJar\Provider\TaxJarProvider">
+            <tag name="shopware.tax.provider" />
+        </service>
+    </services>
+```
+
+{% endcode %}
+
+## Migrate your tax provider to the database
+
+Last but not least, you will have to create a migration to add your tax provider to the database. This is necessary, so that Shopware knows of your tax provider in the first place during checkout.
```

Comment:
You could create it via repository during the Plugin installation process as well, right? Maybe you can write a sentence, that this is a valid alternative?